### PR TITLE
Add README

### PR DESCRIPTION
**Pull request by soebbel**
Add README.md to repository / image

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,52 @@
-# docker-azure-pipelines-terraform
+# Docker image for running Terraform commands in Azure Pipelines container job
 Docker image for running Terraform commands in an Azure Pipelines container job
```
Comment (pascalberger) [inline README.md:2]: Remove this line, as same information comes after the badges

File: README.md [hunk 2]
```diff
@@ -1,2 +1,52 @@
...
+
+[![License](https://img.shields.io/badge/license-MIT-blue.svg?style=flat-square)](https://github.com/swissgrc/docker-azure-pipelines-terraform/blob/main/LICENSE) [![Build](https://img.shields.io/docker/cloud/build/swissgrc/azure-pipelines-terraform.svg?style=flat-square)](https://hub.docker.com/r/swissgrc/azure-pipelines-terraform/builds) [![Pulls](https://img.shields.io/docker/pulls/swissgrc/azure-pipelines-terraform.svg?style=flat-square)](https://hub.docker.com/r/swissgrc/azure-pipelines-terraform) [![Stars](https://img.shields.io/docker/stars/swissgrc/azure-pipelines-terraform.svg?style=flat-square)](https://hub.docker.com/r/swissgrc/azure-pipelines-terraform)
+
+Docker image to run Terraform commands in [Azure Pipelines container jobs].
+
+## Usage
+
+This container can be used to run Terraform commands in [Azure Pipelines container jobs].
```
Comment (pascalberger) [inline README.md:10]: I would add here a sentence that the image also comes with Azure CLI installed.

File: README.md [hunk 3]
```diff
@@ -1,2 +1,52 @@
...
+| unstable | Latest unstable release (from `develop` branch)                                         | ![Docker Image Size (tag)](https://img.shields.io/docker/image-size/swissgrc/azure-pipelines-terraform/unstable?style=flat-square) |
+
+### Configuration
+
+These environment variables are supported:
+
+| Environment variable | Default value | Description                 |
+|----------------------|---------------|-----------------------------|
+| AZURECLI_VERSION     | `2.9.1`       | Version of Azure CLI to use.|
+| NODE_VERSION         | `15.8.0`      | Version of Node to use.     |
```
Comment (pascalberger) [inline README.md:48]: For the Helm chart we didn't mention the `NODE_VERSION` and `YARN_VERSION` arguments (as they are in the base image and only used for Azure Pipelines integration). We should either add it for [azure-pipelines-helm](https://github.com/swissgrc/docker-azure-pipelines-helm#configuration) image too, or remove here.

File: README.md [hunk 4]
```diff
@@ -1,2 +1,52 @@
...
+    jobs:
+      - deployment: planTerraform
+        container: swissgrc/azure-pipelines-terraform:latest
+        environment: smarthotel-dev
+        strategy:
+          runOnce:
+            deploy:
+              steps:
+                - bash: |
+                    az login -u $AZURE_USERNAME -p $AZURE_PASSWORD
```
Comment (pascalberger) [inline README.md:29]: I would suggest to show two examples. One just for calling a Terraform command. And a second one which also calls `az login` for a Terraform script which uses the Azure Provider. In the second example when we call `az login` I would explain that it requires two secret variables and then we should also add the `env` block for passing the variables to the example.

**Review by pascalberger — CHANGES_REQUESTED** [review]

**Comment by soebbel** [comment]
@pascalberger ready for next review

File: README.md [hunk 5]
```diff
@@ -1,2 +1,69 @@
...
-Docker image for running Terraform commands in an Azure Pipelines container job
+# Docker image for running Terraform commands in Azure Pipelines container job
+
+[![License](https://img.shields.io/badge/license-MIT-blue.svg?style=flat-square)](https://github.com/swissgrc/docker-azure-pipelines-terraform/blob/main/LICENSE) [![Build](https://img.shields.io/docker/cloud/build/swissgrc/azure-pipelines-terraform.svg?style=flat-square)](https://hub.docker.com/r/swissgrc/azure-pipelines-terraform/builds) [![Pulls](https://img.shields.io/docker/pulls/swissgrc/azure-pipelines-terraform.svg?style=flat-square)](https://hub.docker.com/r/swissgrc/azure-pipelines-terraform) [![Stars](https://img.shields.io/docker/stars/swissgrc/azure-pipelines-terraform.svg?style=flat-square)](https://hub.docker.com/r/swissgrc/azure-pipelines-terraform)
+
+Docker image to run Terraform commands in [Azure Pipelines container jobs].
+
+## Usage
+
+This container can be used to run Terraform commands in [Azure Pipelines container jobs]. For executing Azure command Azure CLI is included as well.
```
Comment (pascalberger) [inline README.md:9]: Typo. Also we should call it image not container (a container is a running instance)

```suggestion
This image can be used to run Terraform commands in [Azure Pipelines container jobs]. For executing Azure commands Azure CLI is included as well.
```

File: README.md [hunk 6]
```diff
@@ -1,2 +1,69 @@
...
+        environment: smarthotel-dev
+        strategy:
+          runOnce:
+            deploy:
+              steps:
+                - bash: |
+                    terraform init -backend-config=./config/dev_backend.tfvars
+```
+
+The following example defines a deployment with the container using az login before calling Terraform commands.
```
Comment (pascalberger) [inline README.md:31]: You can format `az login` as code:

```suggestion
The following example defines a deployment with the container using `az login` before calling Terraform commands.
```

File: README.md [hunk 7]
```diff
@@ -1,2 +1,69 @@
...
+Note that the two secret variables ($AZURE_USERNAME and $AZURE_PASSWORD) are passed into by the env block.
```
Comment (pascalberger) [inline README.md:32]: The variables which are passed are called `Azure.UserName` and `Azure.Password` (`$AZURE_USERNAME` is to get the value of the environment variable inside the bash task)

```suggestion
Note that two secret variables (`Azure.UserName` and `Azure.Password`) are passed to the task by the env block.
```

**Review by pascalberger — CHANGES_REQUESTED** [review]
Some minor suggestions

**Review by pascalberger — APPROVED** [review]
LGTM

**Pull request merged**